### DISCUSSION TITLE
Can Someone Help Me With KeyDown Events in Android?

### DISCUSSION THREAD

**Discussion by seanocali** [unoplatform/uno#10162] [Q&A]
On my Android emulator it has hardware keyboard enabled and it works when typing in a TextBox. But I can't seem to capture events.

**Comment by jeromelaban** [comment]
Can you provide a small sample of what you're trying to do?

**Reply by seanocali** [reply]
```
        public MainPage()
        {
            this.InitializeComponent();
            var lv = new ListView();
            lv.Loaded += Lv_Loaded;
            var tb = new TextBlock { Text = "Item1" };
            lv.Items.Add(tb);
            lv.KeyDown += Lv_KeyDown;
            this.Content = lv;
        }

        private void Lv_KeyDown(object sender, KeyRoutedEventArgs e)
        {
            //Do stuff
        }
```

**Reply by seanocali** [reply]
Is this what you were asking for or something else?

**Reply by seanocali** [reply]
Sorry about previous example. The above works in UWP. Uno documentation about KeyDown for Android is confusing. The table says it's supported (hardware only) but then it says routed events aren't implemented so it won't bubble in managed code. But a different Uno docs article says it won't bubble in native code.

I was watching Martin Zikmund's live coding session doing gamepad support from last year and he was handling KeyDown events on Android. I'm just trying to learn what needs to be done to make it work.

**Reply by jeromelaban** [reply]
It looks like at this time, Uno android is not supporting keyboard events, either through CoreWindow.KeyDown or through UIElement.KeyDown, we're tracking this through https://github.com/unoplatform/uno/issues/6074.

I don't recall all the details, but some platforms don't necessarily make keyboard input easy to access globally (or locally) because of the lack of native "focus" management, aside from text input controls.

Maybe @MartinZikmund has more details.

**Reply by MartinZikmund** [reply]
We definitely need to add support for this, hopefully will get to it soon. Even a basic support for hw keyboard on iOS and Android will go a long way
